conformance/tests/qualifiers_final_decorator.py

```diff
@@ -65,7 +65,7 @@ def method3() -> None:  # E[method3]
         pass
 
     @overload  # E[method4]
-    def method4(self, x: int) -> int:
+    def method4(self, x: int) -> int:  # E[method4]
```

Comment:
Pyrefly sometimes emits diagnostics on the line of the `def`, not the line with the `@overload`

This is similar to what I did in https://github.com/python/typing/pull/1980